src/extension/components/PopupWindow.re

```diff
@@ -0,0 +1,143 @@
+open Core;
+
+let getLangList conversionType =>
+  switch (conversionType |> Js.String.split "to") {
+  | [| inLang, outLang |] => (Some inLang, Some outLang)
+  | _ => (None, None)
+};
+
+module PopupWindow = {
+  include ReactRe.Component.Stateful.InstanceVars;
+  let name = "PopupWindow";
+
+  type props = {
+    initialText: string
+  };
+  type state = {
+    copy: bool,
+    in_: string,
+    out: string,
+    inLang: option string,
+    outLang: option string
+  };
+  type instanceVars = {
+    mutable _in: option ReactRe.reactRef
+  };
+
+  let getInstanceVars () => {
+    _in: None
```

Comment:
The original was named terribly. This could be something like `inputRef` - right now it's not clear how it differs from `state.in_` (also poorly named - but it was cute at the time).